app/styles/_helpers.scss

```diff
@@ -382,129 +397,116 @@ time {
 }
 
 // Common margin overrides
-// --------------------------------------------------
-
-// top
-.sd-margin-t--0 {
-    margin-top: 0 !important;
-}
-.sd-margin-t--1 {
-    margin-top: 1rem !important;
-}
-.sd-margin-t--2 {
-    margin-top: 2rem !important;
-}
-.sd-margin-t--3 {
-    margin-top: 3rem !important;
-}
-// right
-.sd-margin-r--0 {
-    margin-right: 0 !important;
-}
-.sd-margin-r--1 {
-    margin-right: 1rem !important;
-}
-.sd-margin-r--2 {
-    margin-right: 2rem !important;
+// -------------------------------------------------
+
+// $sd-base-increment - default size of the increment is 0.8rem;
+// For example: .sd-margin-t--1 {margin-top: 0.8rem}, .sd-padding-l--2 {padding-left: 1.6rem}
+$sd-increments: ('0': 0, '0-5': 0.5, '1': 1, '1-5': 1.5, '2': 2, '3': 3, '4' : 4, '5' : 5); // Number of increments, genarates new classes eg. .sd-padding-top--4, .sd-margin-top--4
+$sd-sides: (top, right, bottom, left, x, y); // It's generating these sides
+$sd-properties: (padding, margin); // It's generating these properties
+
+@each $incrementName, $incrementValue in $sd-increments {
+    @each $property in $sd-properties {
+        @each $side in $sd-sides {
+            @if($side == x) {
+                .sd-#{$property}-#{$side}--#{$incrementName} {
+                    #{$property}-left: #{$sd-base-increment * $incrementValue} !important;
+                    #{$property}-right: #{$sd-base-increment * $incrementValue} !important;
+                }
+            } @else if ($side == y) {
+                .sd-#{$property}-#{$side}--#{$incrementName} {
+                    #{$property}-top: #{$sd-base-increment * $incrementValue} !important;
+                    #{$property}-bottom: #{$sd-base-increment * $incrementValue} !important;
```

Comment:
You shouldn't use `important` for base styles.

Reply:
Those are not base styles but helpers that need to override all set styles. it's common practice in UI frameworks for helper classes. It has to stay as it is.